### ISSUE TITLE
E2: Unused Boolean variable in sub-tables of "wire_expression2_funcs" on client

### ISSUE THREAD

**Issue by CaptainPRICE** [wiremod/wire#1157]
In short, since I am (also) working on generating (new) more complete and updated wiki for Expression 2, I have come across unused Boolean value on client per sub-table in `wire_expression2_funcs` table.
`wire_expression2_funcs` table seems to be constructed on server and then sent to client, it contains function data and is indexed by a signature of function (string).

- 1 = string : signature
- 2 = string : return type
- 3 = boolean : this is **always set to false for all functions** on clientside
- 4 = number : cost
- argnames = table : argument/parameter names

Here is the first result from `wire_expression2_funcs` table (currently there are 2418 sub-tables/functions in it (Expression 2)):

```
aabbMax(e:):
	1	=	aabbMax(e:)
	2	=	v
	3	=	false
	4	=	10
	argnames:
		1	=	this
```

There you see it now, `3 = false` part is what this is about.
My suggestion is the following, since that is unused variable (note, it is set [on clientside](https://github.com/wiremod/wire/blob/e3092e3f4acc6c1839211ce3b12768f2a576d48e/lua/entities/gmod_wire_expression2/core/init.lua#L285)) - **make use of it** (that is 2418 bytes of wasted memory, well that will *eventually* be freed by Lua garbage collector...).
Make use of it like, set it to true if function is alias/duplicate or is deprecated/obsolete (it would help a little when generating wiki..). :-)

**Edit**:
I should note that on serverside, value at index 3 holds a delegate to that function..
Basically, I am asking to have two Boolean values (add one more on clientside, and two on serverside), make one tell whether a function is alias/duplicate or not, and other one to tell whether a function is deprecated/obsolete or not.
(It would probably be a good idea to have this information on the server table, too..)
**Bonus suggestion**! Once that's done, perhaps add another two new options in E2 Helper to filter the results from alias/duplicate and/or deprecated/obsolete functions..

**Comment by TomyLobo** [comment]
It used to be the case that all the extensions were sent to the clients, even if they didn't have any client-side parts at all.
This complicated the ``e2function`` mechanism (gmod checked the syntax of those files before transferring), which is why I replaced it by simply sending the function signatures.

That's more reliable, faster and uses less memory on the clients.
Now in order to maintain the same structure in that table, I had to pad it with something in place of the function.
``nil`` is a bad idea, since in Lua it can interrupt what ``unpack`` reads from the array (and we use unpack on those arrays a few times), so I picked ``false``
